web/src/compose_validate.ts

```diff
@@ -795,3 +800,124 @@ export function convert_mentions_to_silent_in_direct_messages(
     const silent_mention_text = people.get_mention_syntax(full_name, user_id, true);
     return silent_mention_text;
 }
+
+export function is_split_messages_enabled(): boolean {
+    return split_messages_enabled;
+}
+
+export function set_split_messages_enabled(enable: boolean): void {
+    split_messages_enabled = enable;
+}
+
+/**
+ * @param content
+ * @returns index of the delimiter occurring outside code blocks since no splitting should
+ * be done inside code blocks
+ */
+function delimiter_index_outside_code_block(content: string): number {
+    let index = content.indexOf(SPLIT_DELIMITER);
+    // skip delimiters which occur inside code blocks
```

Comment:
Much more valuable comments would be to explain WHY the product goal is to find delimiters outside code blocks... and why we search for the next occurrence of the delimiter.

Also, please use complete sentences and normal punctuation (start with a capital letter, end with a period) in comments. It is an important habit for the readability of the codebase.

Reply:
Understood, I have made amendments to the comments.